flink-test-utils-parent/flink-connector-test-utils/src/main/java/org/apache/flink/connector/testframe/testsuites/SourceTestSuiteBase.java

```diff
@@ -448,7 +451,7 @@ public void testSourceMetrics(
                                     testEnv.getRestEndpoint(),
                                     jobClient.getJobID()));
 
-            waitUntilCondition(
+            waitUtil(
```

Comment:
What was the actual issue in your test? Was your count to low, or too high?

Reply:
too low (0) because of the bug in my Cassandra source implementation I mentioned (privately): the intermediate type I used in the source did not contain a single record but multiple ones.  

with this timeout fix + bounded test version (https://github.com/apache/flink/pull/21037) `testSourceMetrics` passes on my Cassandra source (PR to come soon)